File: app/src/main/java/mono/hg/preferences/DesktopPreference.java
package mono.hg.preferences;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import mono.hg.R;
import mono.hg.utils.Utils;

public class DesktopPreference extends PreferenceFragmentCompat {
    private Preference.OnPreferenceChangeListener RotatingListListener = new Preference.OnPreferenceChangeListener() {
        @Override public boolean onPreferenceChange(Preference preference, Object newValue) {
            requireActivity().setRequestedOrientation(Integer.parseInt((String) newValue));
            return true;
        }
    };

    @Override public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.pref_desktop, rootKey);
    }

    @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        ListPreference orientationMode = findPreference("orientation_mode");

        orientationMode.setOnPreferenceChangeListener(RotatingListListener);

        // Window bar hiding works only reliably in KitKat and above.
        if (Utils.atLeastKitKat()) {
            findPreference("windowbar_mode").setVisible(true);
        } else {
            findPreference("windowbar_status_switch").setVisible(true);
        }
    }
}

File: app/src/main/java/mono/hg/preferences/DesktopPreference.kt
package mono.hg.preferences

import android.os.Bundle
import android.view.View
import androidx.annotation.Keep
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import mono.hg.R
import mono.hg.utils.Utils

/**
 * Preferences for the desktop/main view of the launcher.
 */
@Keep
class DesktopPreference : PreferenceFragmentCompat() {
    private val RotatingListListener = Preference.OnPreferenceChangeListener { _, newValue ->
        requireActivity().requestedOrientation = Integer.parseInt(newValue as String)
        true
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.pref_desktop, rootKey)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        findPreference<ListPreference>("orientation_mode")?.onPreferenceChangeListener =
            RotatingListListener

        // Window bar hiding works only reliably in KitKat and above.
        if (Utils.atLeastKitKat()) {
            findPreference<Preference>("windowbar_mode")?.isVisible = true
        } else {
            findPreference<Preference>("windowbar_status_switch")?.isVisible = true
        }
    }
}